In [78]:
import numpy as np
from scipy import stats

## Урок 5. Проверка статистических гипотез. Доверительные интервалы. A/B-тестирование.

**1. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.**

Проверяется гипотеза $H_0$ о том что матожидание распределения диаметра шариков не преавышает 17мм. Так как распределение нормальное и с известной дисперсией, то для проверки гипотезы $H_0$ будем использовать статистику $T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}}$. Эта статистика имеет стандартное нормальное распределение. Для нашей выборки оказалось что $\overline{X}=17.5$. $n = 100$, $\sigma = 2мм$, $a = 17мм$. То есть 
$$T(X) = \dfrac{\overline{X} - 17}{2 / \sqrt{100}} = (\overline{X} - 17) \cdot 5$$


При уровне значимости $\alpha = 0.025$ правосторонняя критическая область $\Omega_\alpha = (t_{1 - \alpha}, \infty)$. Посчитаем квантиль $t_{1 - \alpha}$.

In [79]:
alpha = 0.025
t2 = stats.norm.ppf(1 - alpha)
t2

1.959963984540054

Получили правостороннюю критическую область $\Omega_\alpha = (1.96, \infty)$. Посчитаем значение статистики $T(X)$ для $\overline{X} = 17.5$.

In [80]:
T = lambda x: (x - 17) * 5
T(17.5)

2.5

Для нашей конкретной выборки статистика $T(17.5) = 2.5$ попадает в критическую область, поэтому гипотеза $H_0$ отвергается. То есть матожидание распределения шариков превышает 17мм.

**Ответ: гипотеза не верна.**

**2. Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания a с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.**

Задано $\sigma = 16$, $p = 0.95$, $\overline{X} = 80$, $n = 256$. Так как генеральная совокупность распределена нормально, то доверительный интервал оценки матожидания $a$ можно найти по формуле

$$
P \left(
    \overline{X} - t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    \leq a \leq 
    \overline{X} + t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
\right) = p.
$$

Здесь $\alpha = 1 - p = 0.05$. Так как мы работаем со случайной величиной, а не с выборкой, то квантиль $t_{1 - \alpha / 2}$ найдём из стандартного нормльного распределения.

In [76]:
alpha = 0.05
sigm = 16
p = 0.95
mean = 80
n = 256

alpha = 1 - p
t = stats.norm.ppf(1 - alpha/2)

L = mean - t * (sigm / np.sqrt(n))
U = mean + t * (sigm / np.sqrt(n))

L, U

(78.04003601545995, 81.95996398454005)

**Ответ: доверительный интервал для оценки матожидания $P(78.04 \leq a \leq 81.96) = 0.95$.**

**3. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:**

```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```

**Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?**

 Проверим гипотезу $H_0$ о том что матожидание веса пачки печенья равно 200г. Так как веса пачек печенья распределены нормально, то используем статистику $T(X) = \dfrac{\overline{X} - a}{\sigma_X / \sqrt{n}}$. Здесь $a = 200$, $n = 10$.

In [87]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
a = 200
n = len(X)
alpha = 0.01
T = lambda mean, sigm: (mean - a) / (sigm / np.sqrt(n))

Статистика $T(X)$ будет иметь распределение Стьюдента. Критическая область $\Omega_\alpha = (-\infty, t_{\alpha/2, \: n-1}) \cup (t_{1 - \alpha/2, \: n-1}, \infty)$. Найдём квантили $t_{\alpha/2, \: n - 1}$ и $t_{1 - \alpha/2, \: n - 1}$.

In [88]:
t1 = stats.t.ppf(alpha/2, df=n-1)
t2 = stats.t.ppf(1 - alpha/2, n-1)
t1, t2

(-3.24983554401537, 3.2498355440153697)

Получили критическую область $\Omega_\alpha = (-\infty, -3.25) \cup (3.25, \infty)$. Посчитаем $T(X)$ для нашей выборки.

In [89]:
mean_x = X.mean()
sigm_x = X.std(ddof=1)
mean_x, sigm_x

(198.5, 4.453463071962462)

In [90]:
T(mean_x, sigm_x)

-1.0651074037450896

Получили $T(X) = -1.065$. Эта точка не попадает в критическую область, поэтому гипотеза $H_0$ остаётся в силе. То есть утверждение продавца истинно с уровнем значимости 1%.

**Ответ: оснований отклонить гипотезу нет.**